packages/turf-collect/index.ts

```diff
@@ -47,34 +38,24 @@ function collect(
   inProperty: string,
   outProperty: string
 ): FeatureCollection<Polygon> {
-  var rtree = rbush<Entry>(6);
-
-  var treeItems = points.features.map(function (item) {
-    return {
-      minX: item.geometry.coordinates[0],
-      minY: item.geometry.coordinates[1],
-      maxX: item.geometry.coordinates[0],
-      maxY: item.geometry.coordinates[1],
-      property: item.properties?.[inProperty],
-    };
-  });
+  const rtree = rbush(6);
 
-  rtree.load(treeItems);
+  rtree.load(points);
   polygons.features.forEach(function (poly) {
     if (!poly.properties) {
       poly.properties = {};
     }
-    var bbox = turfbbox(poly);
-    var potentialPoints = rtree.search({
-      minX: bbox[0],
-      minY: bbox[1],
-      maxX: bbox[2],
-      maxY: bbox[3],
-    });
+
+    var potentialPoints = rtree.search(poly) as FeatureCollection<Point>;
```

Comment:
I think the rbush constructor has a generic that lets you specify what type of thing will be loaded into the tree, then the search method (and other methods) will automatically return features of that type without needing to cast.

Notice line 52 on the left above previously specified the tree would contain type `Entry`
```
var rtree = rbush<Entry>(6)
```

Here's another example: https://github.com/Turfjs/turf/blob/d72985ce1a577b42340fed5fc70efe8e4bc8b062/packages/turf-line-overlap/index.ts#L54

You can do the same like

Reply:
I'm not sure we can do that here because the search type and initiated rbush type are different, i.e. we ingest points by search with a polygon. It ends up with a type error:

```
 Type '"Polygon"' is not assignable to type '"Point"'.
```